services:
  ultravidz-tracking-api:
    container_name: ultravidz-tracking-api
    build: .
    ports:
      - "5000:5000"
    volumes:
      - ./tracking-data:/app/data
      - ./logs:/app/logs
    environment:
      - DATABASE_FILE=/app/data/visitor_tracking.db
    restart: unless-stopped
    healthcheck:
      disable: true
      #test: ["CMD", "curl", "-f", "http://localhost:5000/health"]
      #interval: 30s
      #timeout: 10s
      #retries: 3
    networks:
      - proxy
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.ultravidz-tracking-api.entrypoints=http"
      - "traefik.http.routers.ultravidz-tracking-api.rule=Host(`api.ultravidz.com`)"
      - "traefik.http.middlewares.ultravidz-tracking-api-https-redirect.redirectscheme.scheme=https"
      - "traefik.http.routers.ultravidz-tracking-api.middlewares=stashapp-https-redirect"
      - "traefik.http.routers.ultravidz-tracking-api-secure.entrypoints=https"
      - "traefik.http.routers.ultravidz-tracking-api-secure.rule=Host(`api.ultravidz.com`)"
      - "traefik.http.routers.ultravidz-tracking-api-secure.tls=true"
      - "traefik.http.routers.ultravidz-tracking-api-secure.service=ultravidz-tracking-api"
      - "traefik.http.services.ultravidz-tracking-api.loadbalancer.server.port=5000"
      - "traefik.docker.network=proxy"



networks:
  proxy:
    external: true
#volumes:
#  tracking-data:
#  logs: